### PR TITLE
Add minimal MySQL version to Scraper interface

### LINKED ISSUE #327: Add minimal MySQL version to Scraper interface
@arvenil's PR #265 introduced `Scraper` interface. Now we propose to expand it to include the following method:
```go
type Scraper interface {
	// Version of MySQL from which scraper is available.
	Version() float64
}
```

It then can be used to skip unsupported versions entirely:

```go
	for _, scraper := range e.scrapers {
		if versionNum < scraper.Version() {
			continue
		}
```

I can work on porting this feature from Percona's fork if you are interested.

### PR DESCRIPTION
**Pull request by AlekSi**
Closes #327.

### CODE REVIEW

**Comment by AlekSi** [comment]
The test fails due to me using `gofmt` from Go 1.11. https://golang.org/doc/go1.10#gofmt specifically recommends against testing this on CI:
> Note that these kinds of minor updates to gofmt are expected from time to time. In general, we recommend against building systems that check that source code matches the output of a specific version of gofmt. For example, a continuous integration test that fails if any code already checked into a repository is not “properly formatted” is inherently fragile and not recommended.

**Comment by SuperQ** [comment]
Yes, you'll have to format with Go 1.10 until we upgrade. I know it's not recommended but we do it as a project for consistency enforcement.

**Comment by AlekSi** [comment]
Rebased on top of the current `master` which uses Go 1.11.
#223 introduced integration testing with Docker Compose on Travis CI, and that PR uses it, but it wasn't enabled in Circle CI configuration. May you help me with this?

**Comment by SuperQ** [comment]
It looks like this needs a rebase.

**Comment by AlekSi** [comment]
PTAL

File: collector/exporter_test.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package collector
 
 import (
 	"context"
+	"database/sql"
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply collector/exporter_test.go]: And https://github.com/prometheus/mysqld_exporter/pull/228#issuecomment-325300502.

[the remark this answers is not in the document]
Comment (SuperQ) [reply collector/exporter_test.go]: I guess that was a mistake on my part. If we want to do integration testing, it should be a separate thing outside of `go test`. Go testing should be self-contained.

See how we do the "end to end" test with the node_exporer[0].

[0]: https://github.com/prometheus/node_exporter/blob/master/end-to-end-test.sh

[the remark this answers is not in the document]
Comment (AlekSi) [reply collector/exporter_test.go]: `-short` flag is not enough? Can we use, for example, environment variable to enable integration tests?

[the remark this answers is not in the document]
Comment (SuperQ) [reply collector/exporter_test.go]: I guess I don't know what the best practices are for our codebase. I will check with the other Prometheus developers.

[the remark this answers is not in the document]
Comment (SuperQ) [reply collector/exporter_test.go]: I suppose `-short` would be good enough. I'll send a PR to update circleci to use short testing, then you can rebase this to pull it in.

[the remark this answers is not in the document]
Comment (AlekSi) [reply collector/exporter_test.go]: Thank you!

**Comment by AlekSi** [comment]
Rebased again.

**Comment by AlekSi** [comment]
Thank you. 😄

**Pull request merged**